astronomer/providers/google/cloud/operators/dataproc.py

```diff
@@ -1,12 +1,75 @@
 """This module contains Google Dataproc operators."""
+import time
 from typing import Any, Dict, Optional
 
 from airflow.exceptions import AirflowException
-from airflow.providers.google.cloud.operators.dataproc import DataprocSubmitJobOperator
+from airflow.providers.google.cloud.hooks.dataproc import DataprocHook
+from airflow.providers.google.cloud.operators.dataproc import (
+    DataprocCreateClusterOperator,
+    DataprocSubmitJobOperator,
+)
 from airflow.utils.context import Context
 
 from astronomer.providers.google.cloud.hooks.dataproc import DataprocHookAsync
-from astronomer.providers.google.cloud.triggers.dataproc import DataProcSubmitTrigger
+from astronomer.providers.google.cloud.triggers.dataproc import (
+    DataprocCreateClusterTrigger,
+    DataProcSubmitTrigger,
+)
+
+
+class DataprocCreateClusterOperatorAsync(DataprocCreateClusterOperator):
+    """
+    Create a new cluster on Google Cloud Dataproc Asynchronously.
+
+    :param polling_interval: Time in seconds to sleep between checks of cluster status
+    """
+
+    def __init__(
+        self,
+        *,
+        polling_interval: float = 5.0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.polling_interval = polling_interval
+
+    def execute(self, context: "Context") -> None:
+        """Call create cluster API and defer to DataprocCreateClusterTrigger to check the status"""
+        hook = DataprocHook(gcp_conn_id=self.gcp_conn_id)
+        hook.create_cluster(
+            region=self.region,
+            project_id=self.project_id,
+            cluster_name=self.cluster_name,
+            cluster_config=self.cluster_config,
+            labels=self.labels,
+            request_id=self.request_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        self.defer(
+            trigger=DataprocCreateClusterTrigger(
+                project_id=self.project_id,
+                region=self.region,
+                cluster_name=self.cluster_name,
+                end_time=time.monotonic() + self.timeout,
+                metadata=self.metadata,
+                gcp_conn_id=self.gcp_conn_id,
+                polling_interval=self.polling_interval,
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(self, context: Dict[str, Any], event: Optional[Dict[str, str]] = None) -> str:
```

Comment:
this function is not returning any string, its just raising AirflowException, remove the return type

Reply:
This method has changed a lot since the last reviewed